india_stocks_api/brokers/base.py

```
@@ -24,6 +47,53 @@ def create(cls, broker_name: str, **kwargs) -> "BaseBroker":
+    def _ensure_instruments_ready(self, db_path: str = 'instruments.db'):
+        """
+        Check if instruments DB is stale and rebuild if needed.
+        Called automatically by BrokerMeta after __init__.
+        
+        Args:
+            db_path: Path to the instruments database
+        """
+        if self._is_db_stale(db_path):
+            from ..internal import context
+            logger = context.get_logger(__name__)
+            logger.info(f"Instruments DB is stale or missing. Downloading master contract...")
```

Ruff (0.14.13)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)